concordium-std/src/impls.rs

```diff
@@ -800,6 +800,17 @@ impl<T: Serial> SerialCtx for &[T] {
     }
 }
 
+impl<T: Serial> SerialCtx for Vec<T> {
+    fn serial_ctx<W: Write>(
+        &self,
+        size_len: schema::SizeLength,
+        out: &mut W,
+    ) -> Result<(), W::Err> {
+        let slice: &[T] = &self;
+        slice.serial_ctx(size_len, out)
```

Comment:
```suggestion
        self.as_slice().serial_ctx(size_len, out)
```
and remove the `let slice = ...`